Sources/Networking/HTTPClient.swift

```diff
@@ -185,65 +183,42 @@ private extension HTTPClient {
                urlRequest: URLRequest,
                data: Data?,
                error networkError: Error?
-    ) -> Result<HTTPResponse<Data>, Error>? {
+    ) -> HTTPResponse<Data>.Result? {
         if let networkError = networkError {
-            return .failure(networkError)
+            return Result
+                .failure(networkError)
                 .mapError { error in
                     if self.dnsChecker.isBlockedAPIError(networkError),
                        let blockedError = self.dnsChecker.errorWithBlockedHostFromError(networkError) {
                         Logger.error(blockedError.description)
                         return blockedError
                     } else {
-                        return error
+                        return .networkError(error)
                     }
                 }
         }
 
         guard let httpURLResponse = urlResponse as? HTTPURLResponse else {
-            return .failure(ErrorUtils.unexpectedBackendResponseError())
+            return .failure(.unexpectedResponse(urlResponse))
         }
 
         /// - Returns `nil` if status code is 304, since the response will be empty
         /// and fetched from the eTag.
-        func dataIfAvailable(_ statusCode: HTTPStatusCode) throws -> Data? {
+        func dataIfAvailable(_ statusCode: HTTPStatusCode) -> Data? {
             if statusCode == .notModified {
                 return nil
             } else {
                 return data
             }
-
-//            if let data = data, statusCode != .notModified {
-//                let result: [String: Any]?
-//
-//                do {
-//                    result = try JSONSerialization.jsonObject(with: data) as? [String: Any]
-//                } catch let jsonError {
-//                    Logger.error(Strings.network.parsing_json_error(error: jsonError))
-//
-//                    let dataAsString = String(data: data, encoding: .utf8) ?? ""
-//                    Logger.error(Strings.network.json_data_received(dataString: dataAsString))
```

Comment:
Because these logs are potentially important for users to help us debug errors, I've kept them and explicitly logged them when creating `NetworkError.decoding`.